deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -788,7 +784,13 @@ def restore(self, checkpoint=None):
     if checkpoint is None:
       raise ValueError('No checkpoint found')
     with self._get_tf("Graph").as_default():
-      saver = tf.train.Saver()
+      reader = NewCheckpointReader(checkpoint)
+      var_names = set([x for x in reader.get_variable_to_shape_map()])
+      var_map = {
+          x.op.name: x
+          for x in tf.global_variables() if x.op.name in var_names
+      }
+      saver = tf.train.Saver(var_list=var_map)
```

Comment:
If you don't pass in a var map saver.restore throws an exception of your Graph doesn't have the exact same variables as in the checkpoint.  This new restore restores variables if they exist in the checkpoint -- otherwise it leaves the default initializer values from build()